core/include/detray/builders/detail/bin_association.hpp

```diff
@@ -223,9 +223,10 @@ static inline void bin_association(const context_t & /*context*/,
                                     {z_max_pos, constant<scalar_t>::pi});
                                 s_c_pos.push_back(
                                     {z_min_pos, constant<scalar_t>::pi});
-                                surface_contours = {s_c_neg, s_c_pos};
+                                surface_contours.insert(surface_contours.end(),
```

Comment:
Is this also for warning fix? This seems to make a difference in the outputs

Reply:
The assignment seemed to trigger a nullptr warning in gcc 14, why exactly I don't know. The vectors seem to be local to this scope, so I thought it would not change anything. Which outputs do you mean that are different?

Reply:
Seems OK. I thought it is supposed to keep adding elements as surface coutours is in vector but insert or push_back seems happen only once.

Reply:
Yes, I admit, it is less readable now. So far I did not find a better way to silence the warning...